apps/api/src/app/shared/services/cache/invalidate-cache.service.ts

```diff
@@ -0,0 +1,52 @@
+import { CacheService } from './cache.service';
+import { buildKey, CacheInterceptorTypeEnum } from '../../interceptors';
+import { Injectable } from '@nestjs/common';
+
+@Injectable()
+export class InvalidateCacheService {
+  constructor(private cacheService: CacheService) {}
+
+  async execute({
+    storeKeyPrefix,
+    credentials,
+  }: {
+    storeKeyPrefix: CacheKeyPrefixEnum | CacheKeyPrefixEnum[];
+    credentials: Record<string, unknown>;
+  }) {
+    if (!this.cacheService?.cacheEnabled()) return;
+
+    if (Array.isArray(storeKeyPrefix)) {
+      const invalidatePromises = storeKeyPrefix.map((prefix) => {
+        return this.invalidateCase(prefix, credentials);
+      });
+
+      await Promise.all(invalidatePromises);
+    } else {
+      await this.invalidateCase(storeKeyPrefix, credentials);
+    }
+  }
+
+  async invalidateCase(storeKeyPrefix: CacheKeyPrefixEnum, credentials: Record<string, unknown>) {
```

Comment:
It's not really a use case so not sure `execute` is needed here. I would better suggest to define a public API and mark private methods as `private`. Wdyt?

Reply:
Well I think that usecase and execute are just terms but I see that it may confuse in us the near future I will think of another term, regarding the API I totally agree I will update it shortly.